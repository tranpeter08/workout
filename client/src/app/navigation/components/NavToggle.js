import React from 'react';

export default function NavToggle({collapse, toggleNav}) {

  return (
    <button
      name='navButton'
      aria-label={collapse ? "Open Menu" : "Close Menu"}
      onClick={toggleNav}
      className={collapse ? 'nav-button' : 'nav-button open'}>
      <span></span>
      <span></span>
      <span></span>
    </button>
  )
}