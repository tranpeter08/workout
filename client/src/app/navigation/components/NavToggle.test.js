import React from 'react';
import {shallow, mount} from 'enzyme';
import NavToggle from './NavToggle';

describe('<NavTogle />', () => {
  let wrapper;
  it('should render without crashing', () => {
    wrapper = shallow(<NavToggle />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('nav-button')).toBeTruthy();
  })

  it('should have class "open" when collapse is false', () => {
    wrapper = shallow(<NavToggle collapse={false} />)
    expect(wrapper.hasClass('open')).toBeTruthy();
  })

  it('should call "toggleNav" on "click"', () => {
    let toggleNav = jest.fn()
    wrapper = shallow(<NavToggle toggleNav={toggleNav} />);
    wrapper.simulate('click');
    expect(toggleNav).toHaveBeenCalled();
  })
})