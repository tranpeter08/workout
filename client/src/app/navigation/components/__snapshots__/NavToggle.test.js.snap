// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavTogle /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavToggle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Close Menu",
      "children": Array [
        <span />,
        <span />,
        <span />,
      ],
      "className": "nav-button open",
      "name": "navButton",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Close Menu",
        "children": Array [
          <span />,
          <span />,
          <span />,
        ],
        "className": "nav-button open",
        "name": "navButton",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
